### PR TITLE
Update PreviousEmployment reference and Table name

### PR DESCRIPTION
**Pull request by mohinid**
## Notion card
https://www.notion.so/saeloun/Add-tables-for-personal-employment-Allocated-devices-compensation-71ef49d0c38e4242bd2998120141372d

## Summary

1. Removed association between EmploymentDetail and PreviousEmploymentDetail.
2. Renamed PreviousEmploymentDetail to PreviousEmployment to be consistent in naming.
3. Updated RSpecs.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
Tested on local Rails console as well as RSpecs

### Checklist:

- [x] I have manually tested all workflows
- [x] I have performed a self-review of my own code
- [x] I have added automated tests for my code

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -213,7 +214,7 @@
     t.bigint "user_id", null: false
     t.bigint "project_id", null: false
     t.float "duration", null: false
-    t.text "note"
+    t.text "note", default: ""
```
Comment (keshavbiswa) [inline db/schema.rb:217]: I don't think this is part of your db changes.

File: spec/models/previous_employment_spec.rb [hunk 2]
```diff
@@ -2,13 +2,12 @@
 
 require "rails_helper"
 
-RSpec.describe PreviousEmploymentDetail, type: :model do
-  subject(:previous_employment_detail) { build(:previous_employment_detail) }
+RSpec.describe PreviousEmployment, type: :model do
+  subject(:previous_employment) { build(:previous_employment) }
```
[the remark this answers is not in the document]
Comment (mohinid) [reply spec/models/previous_employment_spec.rb]: I think we can name or use just the subject. Both works the same. But not sure which one is the preferred way.

**Review by akhilgkrishnan — APPROVED** [review]

**Review by apoorv-mishra — APPROVED** [review]
LGTM 🎉

**Review by rohitjoshixyz — APPROVED** [review]
LGTM 🎉

**Review by keshavbiswa — APPROVED** [review]
LGTM

**Pull request merged**